lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -124,31 +108,62 @@ def redcap_det_uw_reopening(*, db: DatabaseSession, cache: TTLCache, det: dict,
         LOG.debug("The participant is < 18 years old and we do not have parental consent. Skipping record.")
         return None
 
-    patient_entry, patient_reference = create_patient(enrollment)
-    birthdate = parse_birth_date(enrollment)
+    # Create the participant resource entry and reference.
+    # Assumes that the project language is the participant's preferred language
+    netid = normalize_net_id(enrollment.get('netid'))
+
+    if netid:
+        patient_entry, patient_reference = create_patient_using_unique_identifier(
+            sex = enrollment['core_sex'],
+            preferred_language = LANGUAGE_CODE[enrollment.project.id],
+            unique_identifier = netid,
+            record = enrollment,
+            system_identifier = INTERNAL_SYSTEM)
+    else:
+        patient_entry, patient_reference = create_patient_using_demographics(
+            sex = enrollment['core_sex'],
+            preferred_language = LANGUAGE_CODE[enrollment.project.id],
+            first_name = enrollment['core_participant_first_name'],
+            last_name = enrollment['core_participant_last_name'],
+            birth_date = enrollment['core_birthdate'],
+            zipcode = enrollment['core_zipcode'],
+            record = enrollment,
+            system_identifier = INTERNAL_SYSTEM)
```

Comment:
Partnered with my other comment, this can be single function call with `_create_patient`. If `netid` is None, then the `_create_patient` function will do the right thing of falling back to using the demographic data.

Reply:
Same note as above--for usability. The caller doesn't need to think about first_name, last_name, birth_date, or zipcode if they're creating with a unique identifier. No need to pass in that bunch of parameters with None.

Reply:
@joverlee521 makes a good point here (caveat: I haven't tested this locally).
We can keep the block written here:
```py
if netid:
    ...
else:
    ...
```

and have each block call `_create_patient`. The function call can still look the same (we don't need to pass `first_name`, `last_name` etc in the first block if we rewrite `_create_patient` to have default values of `None` for these arguments. That might be nicer instead of explicitly passing `None` values in each situation, as `create_patient_using_unique_identifier` and `create_patient_using_demographics` currently do.

Reply:
Thanks for the suggestion. I prefer having the two specific overloads.